test/anvil/entrypointsimulations.test.ts

```diff
@@ -0,0 +1,252 @@
+import { TransactionRequest } from '@ethersproject/abstract-provider'
+import { ethers, network } from 'hardhat'
+import { expect } from 'chai'
+
+import {
+  EntryPoint,
+  EntryPointSimulations__factory,
+  SimpleAccount,
+  SimpleAccountFactory
+} from '../../typechain'
+import { EntryPointSimulationsInterface } from '../../typechain/contracts/core/EntryPointSimulations'
+import {
+  checkForBannedOps,
+  createAccount,
+  createAccountOwner,
+  createAddress,
+  deployEntryPoint,
+  fund, getAccountAddress, getAccountInitCode, getBalance, ONE_ETH,
+  simulationResultCatch
+} from '../testutils'
+
+import EntryPointSimulations from '../../artifacts/contracts/core/EntryPointSimulations.sol/EntryPointSimulations.json'
+import { fillAndSign, simulateValidation } from '../UserOp'
+import { BigNumber, Wallet } from 'ethers'
+import { hexConcat } from 'ethers/lib/utils'
+
+// note: to check that the "code override" is properly supported by a node, see if this code returns '0xaa'
+// { code: '0x60aa60005260206000f3' }
+// 0000    60  PUSH1 0xaa
+// 0002    60  PUSH1 0x00
+// 0004    52  MSTORE
+// 0005    60  PUSH1 0x20
+// 0007    60  PUSH1 0x00
+// 0009    F3  *RETURN
+
+describe('EntryPointSimulations', function () {
+  const ethersSigner = ethers.provider.getSigner()
+
+  let account: SimpleAccount
+  let accountOwner: Wallet
+  let simpleAccountFactory: SimpleAccountFactory
+
+  let entryPoint: EntryPoint
+  let entryPointSimulations: EntryPointSimulationsInterface
+
+  before(async function () {
+    if (network.name !== 'anvil') {
+      this.skip()
+    }
+    entryPoint = await deployEntryPoint()
+    entryPointSimulations = EntryPointSimulations__factory.createInterface()
+
+    accountOwner = createAccountOwner();
+    ({
+      proxy: account,
+      accountFactory: simpleAccountFactory
+    } = await createAccount(ethersSigner, await accountOwner.getAddress(), entryPoint.address))
+  })
+
+  it('should use state diff when running the simulation', async function () {
```

Comment:
this is actually a "sanity check" that the tests are valid. so maybe call it from the "before", not as a separate test.